lib/api/services/wallet_service.dart

```diff
@@ -243,68 +183,152 @@ class WalletService extends ApiService {
-                  amount: t.amount, fee:transactions.fee!,
-                  baseAddress: walletAddress
-              );
-              transactionsList.add(transaction);
-            }
+            TransactionListResponse.fromJson(jsonDecode(transactionsData.body));
+
+        List<Transaction> transactionsList = [];
+        for (TransactionGroupResponse transactions
+            in transactionListResponse.transactions) {
+          for (TransactionsResponse t in transactions.transactions) {
+            Transaction transaction = Transaction(
+                type: transactions.type,
+                timestamp: transactions.timestamp,
+                block: transactions.block,
+                address: ((t.sender != null) ? t.sender! : t.destination!),
+                amount: t.amount,
+                fee: transactions.fee,
+                baseAddress: walletAddress);
+            transactionsList.add(transaction);
           }
-          return TransactionsGroup(address: walletAddress, transactionsList: transactionsList);
-
+        }
+        return TransactionsGroup(
+            address: walletAddress, transactionsList: transactionsList);
       } else {
-
-        throw Exception('${transactionsData.body}');
+        throw Exception(transactionsData.body);
       }
     } on Exception catch (e) {
       throw Exception(e.toString());
     }
   }
 
-
-
-  // @POST("/v1/send")
   Future<dynamic> sendXCH(
       {required String privateKey,
-      required var amount,
-      required String address,required int fee}) async {
+      required String address,
+      required int amount,
+      required int fee}) async {
+    var totalAmount = amount + fee;
+    var privateKeyObject = PrivateKey.fromBytes(
+        Uint8List.fromList(const HexDecoder().convert(privateKey)));
+    var publicKeyObject = privateKeyObject.getG1();
+    var wallet = walletPuzzle.curry([Program.atom(publicKeyObject.toBytes())]);
+    var puzzleHash = wallet.hash();
+    var address = segwit.encode(Segwit('xch', puzzleHash));
     try {
-      final responseData = await http.post(
-        Uri.parse('${baseURL}/v1/send'),
-        headers: <String, String>{
-          'Content-Type': 'application/json; charset=UTF-8',
-        },
-        body: jsonEncode(<String, dynamic>{
-          'private_key': privateKey,
-          'amount': amount,
-          'destination': address,
-          'fee': fee
-        }),
-      );
-
-      print(
-          'RESPONSE: ${responseData.body.toString()}  ${responseData.statusCode}');
+      final responseData = await http.post(Uri.parse('$baseURL/v2/records'),
+          headers: <String, String>{
+            'Content-Type': 'application/json; charset=UTF-8',
+          },
+          body: jsonEncode(<String, dynamic>{
+            'address': address,
+          }));
       if (responseData.statusCode == 200) {
-        return 'success';
+        RecordsResponse recordsResponse =
```

Comment:
There's a lot going on here, 
1. Can lines 238-258 be put into a separate function?
2. Can lines 261-306 be put into a separate function?

Reply:
Yes, I'll move these out of this function for clarity.